products/deploymentmanager/terraform.yaml

```diff
@@ -0,0 +1,73 @@
+# Copyright 2017 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Provider::Terraform::Config
+overrides: !ruby/object:Overrides::ResourceOverrides
+  Deployment: !ruby/object:Overrides::Terraform::ResourceOverride
+    id_format: "projects/{{project}}/deployments/{{name}}"
+    import_format: ["projects/{{project}}/deployments/{{name}}"]
+    timeouts: !ruby/object:Api::Timeouts
+      # Very long to support very long deployments
+      insert_minutes: 60
+      update_minutes: 60
+      delete_minutes: 60
+    docs: !ruby/object:Provider::Terraform::Docs
+      warning: |
+        Deployment Manager shares similar behavior with Terraform as both
+        products manage GCP resource lifecycle and state. This Terraform
+        resource is intended only to manage a Deployment resource,
+        and attempts to manage the Deployment's resources in Terraform as well
+        will likely result in errors or unexpected behavior as the two tools
+        fight over ownership. We strongly discourage doing so unless you are an
+        experienced user of both tools.
+    examples:
+      - !ruby/object:Provider::Terraform::Examples
+        name: "deployment_manager_deployment_basic"
+        primary_resource_id: "deployment"
+        primary_resource_name: "fmt.Sprintf(\"my-deployment-%s\", context[\"random_suffix\"])"
+        vars:
+          deployment_name: "my-deployment"
+        skip_test: true
```

Comment:
Did applying files in tests end up being infeasible?

Reply:
More that it was, imo, about the same work and custom code as a handwritten test would create, and so I didn't see much benefit to having the codegen test particularly for this? I know I was the one who made the testCustomContext, but I talked about it with chrisst at some point about how it's a bit unintuitive to figure out that some things just have special substitutions hidden in a ruby file somewhere that makes it work. 

This also lets me use the custom destroy method